core/src/inputs/advice.rs

```diff
@@ -72,11 +63,11 @@ impl AdviceSet {
     /// - The specified depth is greater than the depth of this advice set.
     /// - The specified index is invalid in the context of the specified depth.
     /// - This advice set does not contain a node at the specified index and depth.
-    pub fn get_node(&self, depth: u32, index: u64) -> Result<Word, AdviceSetError> {
+    pub fn get_node(&self, index: NodeIndex) -> Result<Word, MerkleError> {
```

Comment:
We should probably update doc comments for this and similar functions as they still refer to `depth` parameter.

Reply:
Docs updated! I also delegated the documentation to the concrete implementations. Instead of attempting to enum all the possible errors (which will necessarily fail, if we change `miden-crypto`, I put a link to the documentation that will always be updated.